realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -80,48 +86,81 @@ public void call() {
     @Override
     public Observable<DynamicRealm> from(final DynamicRealm realm) {
         checkRxJavaAvailable();
+        final RealmConfiguration realmConfig = realm.getConfiguration();
         return Observable.create(new Observable.OnSubscribe<DynamicRealm>() {
             @Override
             public void call(final Subscriber<? super DynamicRealm> subscriber) {
+                // Get instance to make sure that the Realm is open for as long as the
+                // Observable is subscribed to.
+                final DynamicRealm observableRealm = DynamicRealm.getInstance(realmConfig);
                 final RealmChangeListener listener = new RealmChangeListener() {
                     @Override
                     public void onChange() {
                         if (!subscriber.isUnsubscribed()) {
-                            subscriber.onNext(realm);
+                            subscriber.onNext(observableRealm);
                         }
                     }
                 };
-                realm.addChangeListener(listener);
+                observableRealm.addChangeListener(listener);
                 subscriber.add(Subscriptions.create(new Action0() {
                     @Override
                     public void call() {
-                        realm.removeChangeListener(listener);
+                        observableRealm.removeChangeListener(listener);
+                        observableRealm.close();
                     }
                 }));
 
                 // Immediately call onNext with the current value, as due to Realms auto-update, it will be the latest
                 // value.
-                subscriber.onNext(realm);
+                subscriber.onNext(observableRealm);
             }
         });
     }
 
     @Override
-    public <E extends RealmObject> Observable<RealmResults<E>> from(Realm realm, RealmResults<E> results) {
+    public <E extends RealmObject> Observable<RealmResults<E>> from(final Realm realm, final RealmResults<E> results) {
```

Comment:
I don't think it is custom to have final parameters in interfaces? Mostly because the only reason for doing so is because of implementation specific details which the interface does not care about.